src/meta/ee/src/lib.rs

```diff
@@ -11,3 +11,111 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+use std::sync::Arc;
+use std::sync::Mutex;
+use std::time::Duration;
+
+use databend_common_base::display::display_unix_epoch::DisplayUnixTimeStampExt;
+use databend_common_license::license::LicenseInfo;
+use jwt_simple::algorithms::ECDSAP256KeyPairLike;
+use jwt_simple::algorithms::ECDSAP256PublicKeyLike;
+use jwt_simple::algorithms::ES256KeyPair;
+use jwt_simple::algorithms::ES256PublicKey;
+use jwt_simple::claims::Claims;
+use jwt_simple::claims::JWTClaims;
+use jwt_simple::prelude::Clock;
+
+#[derive(Clone, Default)]
+pub struct MetaServiceEnterpriseGate {
+    /// License ES256 signed jwt claim token.
+    license_token: Arc<Mutex<Option<String>>>,
+
+    /// The public key to verify the license token.
+    public_key: String,
+}
+
+impl MetaServiceEnterpriseGate {
+    const LICENSE_PUBLIC_KEY: &'static str = r#"-----BEGIN PUBLIC KEY-----
```

Comment:
why use a hard code public key?

Reply:
This way only the the license token published by us with our private key will be correctly decoded.
If to let user config public key in config, user will be able to use a self signed token with its private key and let databend-meta decoded it with the public key it provided.